iwyu_path_util.cc

```diff
@@ -167,18 +169,46 @@ string MakeAbsolutePath(const string& path) {
 string MakeAbsolutePath(const string& base_path, const string& relative_path) {
   llvm::SmallString<128> absolute_path(base_path.c_str());
   llvm::sys::path::append(absolute_path, relative_path);
-
-  return absolute_path.str();
+  
+  std::string result = absolute_path.str();
+  StripRight(&result, "./");
+  return result;
 }
 
 string GetParentPath(const string& path) {
   llvm::StringRef parent = llvm::sys::path::parent_path(path);
-  return parent.str();
+  string result = parent.str();
+  // Ensure trailing slash
+  if(!result.empty() && result[result.size() - 1] != '/')
+      result += '/';
+  return result;
+}
+
+bool StripPathPrefix(string* path, const string& prefix_path) {
```

Comment:
I think this is possible, if we make sure, this is only called with absolute, normalized paths (which I think should be done anyway)